crates/sui-config/src/genesis.rs

```diff
@@ -280,7 +398,92 @@ impl Builder {
             &validators,
         );
 
+        let (genesis_transaction, genesis_effects, objects) = create_genesis_transaction(objects);
+        let (checkpoint, checkpoint_contents) =
+            create_genesis_checkpoint(&genesis_transaction, &genesis_effects);
+
+        self.built_genesis = Some((
+            checkpoint,
+            checkpoint_contents,
+            genesis_transaction,
+            genesis_effects,
+            objects,
+        ));
+
+        self.built_genesis.clone().unwrap()
+    }
+
+    fn committee(objects: &[Object]) -> Committee {
+        let sui_system_object = objects
+            .iter()
+            .find(|o| o.id() == sui_types::SUI_SYSTEM_STATE_OBJECT_ID)
+            .expect("Sui System State object must always exist");
+        let move_object = sui_system_object
+            .data
+            .try_as_move()
+            .expect("Sui System State object must be a Move object");
+        let result = bcs::from_bytes::<SuiSystemState>(move_object.contents())
+            .expect("Sui System State object deserialization cannot fail");
+        result.get_current_epoch_committee().committee
+    }
+
+    pub fn build(mut self) -> Genesis {
+        let (checkpoint, checkpoint_contents, transaction, effects, objects) =
+            self.build_unsigned_genesis_data();
+
+        let committee = Self::committee(&objects);
+
+        let transaction = {
+            let signatures = self
+                .signatures
+                .clone()
+                .into_iter()
+                .map(|(_, (_, s))| s)
+                .collect();
+
+            CertifiedTransaction::new(transaction.into_data(), signatures, &committee).unwrap()
+        };
+
+        let checkpoint = {
+            let signatures = self
+                .signatures
+                .clone()
+                .into_iter()
+                .map(|(_, (s, _))| s)
+                .collect();
+
+            CertifiedCheckpointSummary {
+                summary: checkpoint,
+                auth_signature: AuthorityWeakQuorumSignInfo::new_from_auth_sign_infos(
+                    signatures, &committee,
+                )
+                .unwrap(),
+            }
+        };
+
+        let validators = self
+            .validators
+            .into_iter()
+            .map(|(_, v)| v)
+            .collect::<Vec<_>>();
+
+        // Ensure we have signatures from all validators
+        assert_eq!(checkpoint.auth_signature.len(), validators.len() as u64);
+
+        //TODO remove this and don't override previous txn digest
```

Comment:
was this supposed to be temporary or did you mean to merge? (the formatting makes it look unintentional)

Reply:
Yes this is intentional and will be cleaned up in a follow up PR.